### PR TITLE
Model Update: Scenes and Character

### PR DESCRIPTION
**Pull request by Vassyli**
Adds basic Scene models
Extends Character model by providing properties that can be any field:value relationship. The value is stored serialized, meaning it's PHP type is preserved.
Property management can be reused since it's written as a trait: Properties are providing getter and setter for property name and property value. PropertyManager has to be used by the property parent and provides an easy interface to access the properts using getProperty($fieldname, $defaultvalue).

### CODE REVIEW

File: tests/Models/CharacterModelTest.php [hunk 1]
```diff
@@ -104,4 +108,30 @@ public function testDeletion() {
...
+        $firstCharacter = $em->getRepository(Character::class)->find(1);
+        $this->assertSame(5, $firstCharacter->getProperty("dragonkills", 5));
+        $this->assertNotSame(5, $firstCharacter->getProperty("dragonkills", "5"));
+        $this->assertSame("hanniball", $firstCharacter->getProperty("petname", "hanniball"));
+        
+        // test setting variables, then getting
+        $firstCharacter->setProperty("dragonkills", 5);
+        $this->assertSame(5, $firstCharacter->getProperty("dragonkills"));
+        $this->assertNotSame("5", $firstCharacter->getProperty("dragonkills"));
+
```
Comment (austenmc) [inline tests/Models/CharacterModelTest.php:133]: This is good but it doesnt test writing it to the database and reading it back, correct? Just writing and reading from the in-memory properties instance variable.

File: src/Tools/Model/Properties.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    
+    /**
+     * Sets the name of the property
+     * @param string $name
+     * @throws ArgumentEmptyException If parameter $name is empty
+     */
+    public function setName(string $name)
+    {
+        if($name === "") {
```
Comment (austenmc) [inline src/Tools/Model/Properties.php]: nit: spacing

File: src/Tools/Model/Properties.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?php
+declare(strict_types=1);
+
+namespace LotGD\Core\Tools\Model;
+
+const PROPERTY_STRING = 0;
+const PROPERTY_INT = 1;
+const PROPERTY_FLOAT = 2;
+
```
Comment (austenmc) [inline src/Tools/Model/Properties.php]: Where do you use these?

**Comment by austenmc** [comment]
lgtm except for the small comments I made.

File: tests/Models/CharacterModelTest.php (hunk 1 above)
Comment (Vassyli) [inline tests/Models/CharacterModelTest.php:133]: Yes. I need to at a flash procedure and probably check some parts of the sql.
Reading it back is probably difficult due to the truncation before every test.

File: src/Tools/Model/Properties.php (hunk 3 above)
Comment (Vassyli) [inline src/Tools/Model/Properties.php]: Nowhere. I thought I deleted them.

**Pull request merged**